Suppress error formatting for log lines containing certain strings

robertknight opened the pull request:
Adjust the formatting of output from `make dev` to avoid using the bold
red "error" formatting for log lines that contain certain strings, even
if they were written to stderr.

There are some other minor changes in this commit as a result of
formatting the file with Black.

Slack context: https://hypothes-is.slack.com/archives/C1MA4E9B9/p1595931093420100

An alternative way to avoid the problem here would be to change the services producing the offending log lines to write them to stdout instead.

**Before:**

<img width="600" alt="before" src="https://user-images.githubusercontent.com/2458/88683158-6ba3a480-d0eb-11ea-8798-e98cd374a534.png">

**After:**

<img width="600" alt="Screenshot 2020-07-28 at 15 58 21" src="https://user-images.githubusercontent.com/2458/88682980-36975200-d0eb-11ea-84fc-0c765d2ac23b.png">

seanh commented on bin/logger:
```diff
@@ -135,25 +135,36 @@ def _write(s):
-        output_line = prefix + line + '\n'
+        format_as_error = err
+        for marker in NON_ERROR_MARKERS:
+            format_as_error = format_as_error and (marker not in line)
```
I think this is probably a good use-case for Python's `any()` or `all()` builtin

seanh commented:
@robertknight Can you also send the same PR to lms? https://github.com/hypothesis/lms/blob/master/bin/logger

robertknight commented:
PR build failed with:

```
=================================== FAILURES ===================================
_ TestCreateOrUpdateDocumentMeta.test_it_updates_an_existing_DocumentMeta_if_there_is_one _

self = <document_test.TestCreateOrUpdateDocumentMeta object at 0x7fa2b5e420f0>
db_session = <sqlalchemy.orm.session.Session object at 0x7fa2b64d3cf8>

    def test_it_updates_an_existing_DocumentMeta_if_there_is_one(self, db_session):
        claimant = "http://example.com/claimant"
        type_ = "title"
        value = "the title"
        document_ = document.Document()
        created = yesterday()
        updated = now()
        document_meta = document.DocumentMeta(
            claimant=claimant,
            type=type_,
            value=value,
            document=document_,
            created=created,
            updated=updated,
        )
        db_session.add(document_meta)
    
        new_updated = now()
        document.create_or_update_document_meta(
            session=db_session,
            claimant=claimant,
            type=type_,
            value="new value",
            document=document.Document(),  # This should be ignored.
            created=now(),  # This should be ignored.
>           updated=new_updated,
        )

tests/h/models/document_test.py:543: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
h/models/document.py:388: in create_or_update_document_meta
    document.id,
/usr/local/lib/python3.6/logging/__init__.py:1320: in warning
    self._log(WARNING, msg, args, **kwargs)
/usr/local/lib/python3.6/logging/__init__.py:1444: in _log
    self.handle(record)
/usr/local/lib/python3.6/logging/__init__.py:1454: in handle
    self.callHandlers(record)
/usr/local/lib/python3.6/logging/__init__.py:1516: in callHandlers
    hdlr.handle(record)
/usr/local/lib/python3.6/logging/__init__.py:865: in handle
    self.emit(record)
.tox/tests/lib/python3.6/site-packages/_pytest/logging.py:326: in emit
    super().emit(record)
/usr/local/lib/python3.6/logging/__init__.py:1000: in emit
    self.handleError(record)
/usr/local/lib/python3.6/logging/__init__.py:994: in emit
    msg = self.format(record)
/usr/local/lib/python3.6/logging/__init__.py:840: in format
    return fmt.format(record)
.tox/tests/lib/python3.6/site-packages/_pytest/logging.py:89: in format
    return super().format(record)
/usr/local/lib/python3.6/logging/__init__.py:577: in format
    record.message = record.getMessage()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <LogRecord: h.models.document, 30, /var/lib/hypothesis/h/models/document.py, 388, "Found DocumentMeta (id: %d)'s document_id (%d) doesn't match given Document's id (%d)">

    def getMessage(self):
        """
        Return the message for this LogRecord.
    
        Return the message for this LogRecord after merging any user-supplied
        arguments with the message.
        """
        msg = str(self.msg)
        if self.args:
>           msg = msg % self.args
E           TypeError: %d format: a number is required, not NoneType

/usr/local/lib/python3.6/logging/__init__.py:338: TypeError
---------------------------- Captured log teardown -----------------------------
ERROR    sqlalchemy.pool.impl.QueuePool:base.py:706 Exception during reset or similar
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 697, in _finalize_fairy
    fairy._reset(pool)
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/pool/base.py", line 885, in _reset
    "Reset agent is not active.  "
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/util/langhelpers.py", line 1487, in warn
    warnings.warn(msg, exc.SAWarning, stacklevel=2)
sqlalchemy.exc.SAWarning: Reset agent is not active.  This should not occur unless there was already a connectivity error in progress.
__ TestUpdateDocumentMetadata.test_it_uses_the_target_uri_to_get_the_document __

self = <document_test.TestUpdateDocumentMetadata object at 0x7fa2b6378048>
annotation = <Mock spec='Annotation' id='140336818348384'>
Document = <MagicMock name='Document' spec='Document' id='140336818348104'>
session = <Mock spec='Session' id='140336817599824'>

    def test_it_uses_the_target_uri_to_get_the_document(
        self, annotation, Document, session
    ):
        document_uri_dicts = [
            {
                "uri": "http://example.com/example_1",
                "claimant": "http://example.com/claimant",
                "type": "type",
                "content_type": None,
            },
            {
                "uri": "http://example.com/example_2",
                "claimant": "http://example.com/claimant",
                "type": "type",
                "content_type": None,
            },
            {
                "uri": "http://example.com/example_3",
                "claimant": "http://example.com/claimant",
                "type": "type",
                "content_type": None,
            },
        ]
    
        document.update_document_metadata(
            session,
            annotation.target_uri,
            [],
            document_uri_dicts,
            annotation.created,
>           annotation.updated,
        )

tests/h/models/document_test.py:854: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
h/models/document.py:501: in update_document_metadata
    **document_uri_dict
h/models/document.py:305: in create_or_update_document_uri
    document.id,
/usr/local/lib/python3.6/logging/__init__.py:1320: in warning
    self._log(WARNING, msg, args, **kwargs)
/usr/local/lib/python3.6/logging/__init__.py:1444: in _log
    self.handle(record)
/usr/local/lib/python3.6/logging/__init__.py:1454: in handle
    self.callHandlers(record)
/usr/local/lib/python3.6/logging/__init__.py:1516: in callHandlers
    hdlr.handle(record)
/usr/local/lib/python3.6/logging/__init__.py:865: in handle
    self.emit(record)
.tox/tests/lib/python3.6/site-packages/_pytest/logging.py:326: in emit
    super().emit(record)
/usr/local/lib/python3.6/logging/__init__.py:1000: in emit
    self.handleError(record)
/usr/local/lib/python3.6/logging/__init__.py:994: in emit
    msg = self.format(record)
/usr/local/lib/python3.6/logging/__init__.py:840: in format
    return fmt.format(record)
.tox/tests/lib/python3.6/site-packages/_pytest/logging.py:89: in format
    return super().format(record)
/usr/local/lib/python3.6/logging/__init__.py:577: in format
    record.message = record.getMessage()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <LogRecord: h.models.document, 30, /var/lib/hypothesis/h/models/document.py, 305, "Found DocumentURI (id: %d)'s document_id (%d) doesn't match given Document's id (%d)">

    def getMessage(self):
        """
        Return the message for this LogRecord.
    
        Return the message for this LogRecord after merging any user-supplied
        arguments with the message.
        """
        msg = str(self.msg)
        if self.args:
>           msg = msg % self.args
E           TypeError: %d format: a number is required, not Mock

/usr/local/lib/python3.6/logging/__init__.py:338: TypeError
=========================== short test summary info ============================
FAILED tests/h/models/document_test.py::TestCreateOrUpdateDocumentMeta::test_it_updates_an_existing_DocumentMeta_if_th
```

robertknight commented:
The failing tests are unrelated to this branch, which modifies a script not used as part of test execution. See https://github.com/hypothesis/h/pull/6127. On that basis I'm going to merge this anyway.
